src/miner.cpp

```diff
@@ -18,9 +18,12 @@
 #include "util.h"
 #include "utilmoneystr.h"
 #ifdef ENABLE_WALLET
+#include "crypto/equihash.h"
```

Comment:
Yes this is intentional, because all the mining code is itself inside an `#ifdef ENABLE_WALLET` upstream.